src/Moq/Language/Flow/SetupPhrase.cs

```diff
@@ -159,6 +159,114 @@ public IThrowsResult Throws<TException>() where TException : Exception, new()
 			return this;
 		}
 
+		public IThrowsResult Throws(Delegate exceptionFunction)
+		{
+			this.Setup.SetThrowComputedExceptionBehavior(exceptionFunction);
+			return this;
+		}
+
+		public IThrowsResult Throws<TException>(Func<TException> exceptionFunction) where TException : Exception, new()
+		{
+			this.Setup.SetThrowComputedExceptionBehavior(exceptionFunction);
+			return this;
+		}
+		
+		public IThrowsResult Throws<T, TException>(Func<T, TException> valueExpression) where TException : Exception, new()
```

Comment:
`valueExpression` &rarr; `exceptionFunction`? (Same in the other overloads further down.)

Reply:
Fixed in latest commit.